src/CMakeLists.txt

```diff
@@ -1,35 +1,94 @@
 cmake_minimum_required(VERSION 2.8.11)
-project(PSRP_PROVIDER)
+project(PSRP)
 
 # Export commands for auto-completion engines
 set(CMAKE_EXPORT_COMPILE_COMMANDS 1)
+
+# OMI uses some GNU extensions that cannot be found in header files
+# without defining this
 add_definitions(-D_GNU_SOURCE)
 
-# Cannot use add_compile_options with 2.8.11
-set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wall -Werror -fPIC")
+# Dependent on the threading library. Nothing 
+# equivalent for iconv unfortunately
+find_package(Threads REQUIRED)
+
+# Search OpenSSL
+
+if (${CMAKE_SYSTEM_NAME} MATCHES "Darwin")
+	set(OPENSSL_ROOT_DIR /usr/local/opt/openssl)
+	find_package(openssl REQUIRED)
+	message(STATUS "Using OpenSSL ${OPENSSL_VERSION}")
+	message(STATUS "OpenSSL Libraries ${OPENSSL_LIBRARIES}")
+	message(STATUS "OpenSSL Include dirs ${OPENSSL_INCLUDE_DIRS}")
+elseif (ULINUX_SSL EQUAL 1)
+	find_package(PkgConfig REQUIRED)
+	pkg_search_module(OPENSSL REQUIRED openssl)
+	set(OPENSSL_LIBRARIES "-L/usr/local_ssl_1.0.0/lib64 -lssl -lcrypto" )
+	set(OPENSSL_INCLUDES "/usr/local_ssl_1.0.0/inc" )
+	message(STATUS "Using private openssl in ${OPENSSL_LIBRARIES}")
+	message(STATUS "OpenSSL Libraries ${OPENSSL_LIBRARIES}")
+	message(STATUS "OpenSSL Include dirs ${OPENSSL_INCLUDE_DIRS}")
+else ()
```

Comment:
Same here. If it's not Darwin, then isn't it just ULINUX, and that's that? Is there ever a case where we will build for non-ULINUX?